server/paymentProviders/utils.js

```diff
@@ -41,7 +41,8 @@ export const persistTransaction = async (virtualCard, transaction) => {
   const host = virtualCard.host;
   const collective = virtualCard.collective;
   const vendor = await getOrCreateVendor(transaction.vendorProviderId, transaction.vendorName);
-  const hostCurrencyFxRate = await getFxRate('USD', host.currency);
+  const currency = transaction.currency || 'USD';
+  const hostCurrencyFxRate = await getFxRate(currency, host.currency);
   const description = `Virtual Card charge: ${vendor.name}`;
 
   if (transaction.fromAuthorizationId) {
```

Comment:
What does this condition mean actually, could you add a comment?

I know it's not part of the PR, but spotting this while reading the code again.